python/cudf/cudf/core/dataframe.py

```diff
@@ -430,18 +430,67 @@ def mask(self, other):
             df[col]._column = df[col]._column.set_mask(boolbits)
         return df
 
-    def __setitem__(self, name, col):
-        """Add/set column by *name or DataFrame*
+    def __setitem__(self, arg, value):
+        """Add/set column by *arg or DataFrame*
         """
-        if isinstance(name, DataFrame):
+
+        if isinstance(arg, DataFrame):
+            # not handling set_item where arg = df & value = df
             for col_name in self._cols:
-                mask = name[col_name]
-                self._cols[col_name][mask] = col
+                mask = arg[col_name]
```

Comment:
Don't call this mask, this makes me think it's a boolean mask instead of a scatter map.

Reply:
Okay, will change it to `scatter map` or something.

Reply:
Should be resolved with https://github.com/rapidsai/cudf/pull/3442/commits/01b6c8ceec9223948bff9323553f5ec7fc564ecd